src/database/ids.ts

```diff
@@ -0,0 +1,24 @@
+import { getModelForClass, modelOptions, prop } from "@typegoose/typegoose"
+
+export enum IdType {
+	COMMUNITY = "community",
+	RULE = "rule",
+	REPORT = "report",
+	REVOCATION = "revocation",
+	LOG = "log",
+}
+
+@modelOptions({
+	schemaOptions: {
+		collection: "ids",
+	},
+})
+export class IdClass {
+	@prop({ _id: false, unique: true })
```

Comment:
`{id: false}` in a prop is telling typegoose "hey, don't use this as a copy of the `_id` value. It otherwise does that. The same thing is done with all other items that use their own IDs such as reports and revocations.

Reply:
Where is that documented? The documentation I linked says

> Set this to false, if you want to turn off creating IDs for sub-documents.

There is no sub-document involved here.

Reply:
You're right, I thought it was to do with this but I guess that it doesn't work as I thought
```ts
ReportInfoModel.findOne({}).then((report) => {
	if (report) {
		report.id // Any
		report.createdAt // Date
	}
})
```